tests/auth.test.ts

```diff
@@ -97,6 +98,110 @@ describe('account registration', () => {
+      .thenResolve();
+    const authController = ControllerFactory.auth(conn, instance(emailService));
+    const registerRequest = { user };
+    await expect(authController.register(registerRequest, FactoryUtils.randomHexString()))
+      .rejects.toThrow('This handle is already in use.');
+
+    verify(emailService.sendEmailVerification(user.email, user.firstName, anyString()))
+      .never();
+  });
+
+  test('User registration with a long name truncates handle to exactly 32 characters', async () => {
+    const conn = await DatabaseConnection.get();
+    const existingMember = UserFactory.fake();
+
+    await new PortalState()
+      .createUsers(existingMember)
+      .write();
+
+    const user = {
+      email: 'acm@ucsd.edu',
+      firstName: 'ACMACMACMACMACMACMACMACMACMACMACMACMACMACM',
+      lastName: 'UCSDUCSDUCSDUCSDUCSDUCSDUCSDUCSD',
+      password: 'password',
+      major: UserFactory.major(),
+      graduationYear: UserFactory.graduationYear(),
+    };
+
+    // register member
+    const emailService = mock(EmailService);
+    when(emailService.sendEmailVerification(user.email, user.firstName, anyString()))
+      .thenResolve();
+    const authController = ControllerFactory.auth(conn, instance(emailService));
+    const registerRequest = { user };
+    const registerResponse = await authController.register(registerRequest, FactoryUtils.randomHexString());
+
+    expect(registerResponse.user.handle.length).toBe(32);
+  });
+
+  test('User can register with an optional handle to be set', async () => {
+    const conn = await DatabaseConnection.get();
+    const existingMember = UserFactory.fake();
+
+    await new PortalState()
+      .createUsers(existingMember)
+      .write();
+
+    const user = {
+      email: 'acm@ucsd.edu',
+      firstName: 'ACM',
+      lastName: 'UCSD',
+      password: 'password',
+      handle: 'acmadmin',
+      major: UserFactory.major(),
+      graduationYear: UserFactory.graduationYear(),
+    };
+
+    // register member
+    const emailService = mock(EmailService);
+    when(emailService.sendEmailVerification(user.email, user.firstName, anyString()))
+      .thenResolve();
+    const authController = ControllerFactory.auth(conn, instance(emailService));
+    const registerRequest = { user };
+    const registerResponse = await authController.register(registerRequest, FactoryUtils.randomHexString());
+
+    // check that member is registered as expected
+    const params = { handle: registerResponse.user.handle };
+    const getUserResponse = await ControllerFactory.user(conn).getUserByHandle(params, existingMember);
+    expect(getUserResponse.user).toStrictEqual({
+      firstName: user.firstName,
+      lastName: user.lastName,
+      major: user.major,
+      graduationYear: user.graduationYear,
+      bio: null,
+      points: 0,
+      handle: user.handle,
+      uuid: registerResponse.user.uuid,
+      profilePicture: null,
+    });
```

Comment:
Since we only care about the handle in this test, we only need to assert that the handle is what we'd expect it to be, and not the other fields. So let's just assert the handles and omit the other fields.